File: src/java/EntityFX/src/EntityFX/Core/Generic/StringManipulationBase.java
package EntityFX.Core.Generic;

import java.io.FileNotFoundException;

import EntityFX.Core.Writer;

public abstract class StringManipulationBase<TResult> extends BenchmarkBase<TResult> {

    public StringManipulationBase(Writer writer, boolean printToConsole)  throws FileNotFoundException {
        super(writer, printToConsole);
        Iterrations = 5000000;
        Ratio = 10;
    }


    protected static String doStringManipilation(String str) {
        return (String.join("/", str.split(" ")).replace("/", "_").toUpperCase() + "AAA").toLowerCase().replace("aaa", ".");
    }

}

File: src/dotnet/EntityFX.NetBenchamarks.Core/Generic/StringManipulationBase.cs
﻿using System.Runtime.CompilerServices;

namespace EntityFX.NetBenchmark.Core.Generic
{
    public abstract class StringManipulationBase<TResult> : BenchmarkBase<TResult>
    {
        public StringManipulationBase(IWriter writer)
            :base(writer)
        {
            Iterrations = 5000000;
            Ratio = 10;
        }
#if NETSTANDARD2_0 || NET45
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
#endif
        protected static string DoStringManipilation(string str)
        {
            return (string.Join("/", str.Split(' ')).Replace("/", "_").ToUpper() + "AAA").ToLower().Replace("aaa", ".");
        }
    }
}
